### PR TITLE
fix: Incorrect anchor tag

### LINKED ISSUE #5542: Windows anchor links don't work correctly on download/package-manager
### URL:

https://nodejs.org/en/download/package-manager

### Browser Name:

Firefox

### Browser Version:

115.0.2 (64-bit)

### Operating System:

Windows 11, 22H2

### How to reproduce the issue:

- I clicked on the table of contents "Windows" link; Nothing happened.

There are two bugs here:

1. The ToC "Windows" link goes to the anchor `windows-1` instead of the `<h2>` "Windows" anchor.
2. There are two anchors with the `id` "windows".  The first is in "nvs" -> "windows" and the second is "windows".

I don't know what flavor of markdown is being used, so I didn't submit a PR.  A previous MD engine apparently automatically appended numbers to the end.

### PR DESCRIPTION
**Pull request by sohan2410**
## Description

Fix #5542 where the anchor tag was not working for 

- centos-fedora-and-red-hat-enterprise-linux
- windows
- z/OS

## Validation
On clicking, it's scrolling to particular anchor section

## Related Issues
Fixes #5542 

### Check List

- [X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [X] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [X] I have run `npx turbo format` to ensure the code follows the style guide.
- [X] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by mikeesto — COMMENTED** [review]
Thanks for the PR. There is nvs -> Windows, and also Windows. I think Windows from the TOC should link to the Windows at the bottom of the page?

**Comment by sohan2410** [comment]
@mikeesto there's common id (```#windows```) for nvs->windows and windows, shall I rename nvs windows id by ```#windows-nvs```?

**Comment by ovflowd** [comment]
This doesn't really fix the issue. The current problem with ToC comes from our AnchorHeading Component that cannot interpolate duplicate anchors (it will generate the same anchors instead of `anchor-1`, `anchor-2` etc...

So the change should not be done on the Markdown file.

**Comment by sohan2410** [comment]
@ovflowd can you suggest me with the approach!

**Comment by ovflowd** [comment]
I mean, no? I already kinda mentioned that the issue is on `AnchorHeading` component 😅 the problem pretty much is that we use regex to generate the `ids`. I believe this is a poor approach (And the reason why is because we only have the context of the current heading. So if we have headings that in the end, have the same ID, well... They clash).

Pretty much my recommendation is:

1. Remove the AnchorHeading component and its references
2. Add this rehyp plugin to `next.dynamic.mjs` https://github.com/rehypejs/rehype-autolink-headings
  - Pretty much within https://github.com/nodejs/nodejs.org/blob/main/next.dynamic.mjs#L159 (you need to add the `rehypePlugins` property and pass this plugin there.
  - Example: 
    ```ts
    import rehypeAutolinkHeadings from 'rehype-autolink-headings'
    
    ....

    rehypePlugins: [rehypeAutolinkHeadings]
    ```

That's the most guidance I can give you without doing the solution myself. You can validate that this works by checking the link's work. (You need to undo your changes on the Markdown file)

**Comment by sohan2410** [comment]
@ovflowd can you please review

File: next.dynamic.mjs [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { serialize } from 'next-mdx-remote/serialize';
 import * as nextLocales from './next.locales.mjs';
 import * as nextConstants from './next.constants.mjs';
 import * as nextHelpers from './next.helpers.mjs';
+import rehypeAutolinkHeadings from 'rehype-autolink-headings/lib';
```
Comment (ovflowd) [inline next.dynamic.mjs]: Should the import here, really be `/lib`? just out of curiosity!

File: next.dynamic.mjs [hunk 2]
```diff
@@ -9,6 +9,8 @@ import { serialize } from 'next-mdx-remote/serialize';
...
+import rehypeSlug from 'rehype-slug';
```
Comment (ovflowd) [inline next.dynamic.mjs]: What is this plugin and why do we need it?

File: next.dynamic.mjs (hunk 1 above)
Comment (ovflowd) [inline next.dynamic.mjs]: Also, these two imports should be after the remarkHeadings import.

File: package.json [hunk 3]
```diff
@@ -59,6 +59,8 @@
     "prismjs": "~1.29.0",
     "react-icons": "~4.10.1",
     "react-intl": "~6.4.4",
+    "rehype-autolink-headings": "^6.1.1",
```
Comment (ovflowd) [inline package.json]: Please read our Dependency Pinning guidelines :) https://github.com/nodejs/nodejs.org/blob/main/DEPENDENCY_PINNING.md

These should have a `~` version range and not a `^` (both the dependencies you added, and after fixing them, don't forget to run `npm i` again to update the lockfile)

File: providers/mdxProvider.tsx [hunk 4]
```diff
@@ -3,20 +3,12 @@ import { MDXProvider as BaseMDXProvider } from '@mdx-js/react';
...
 
 const mdxComponents: MDXComponents = {
   NodeApiVersionLinks: NodeApiVersionLinks,
-  h1: props => <AnchoredHeading level={1} {...props} />,
-  h2: props => <AnchoredHeading level={2} {...props} />,
-  h3: props => <AnchoredHeading level={3} {...props} />,
-  h4: props => <AnchoredHeading level={4} {...props} />,
-  h5: props => <AnchoredHeading level={5} {...props} />,
-  h6: props => <AnchoredHeading level={6} {...props} />,
-  blockquote: ({ children }) => <div className="highlight-box">{children}</div>,
```
Comment (ovflowd) [inline providers/mdxProvider.tsx:19]: You shouldn't have removed the `blockquote` one, only the h1 to h6 components.

File: next.dynamic.mjs (hunk 1 above)
Comment (sohan2410) [reply next.dynamic.mjs]: Umm I guess VSC auto imported or I might have taken from documentation, will fix that

File: providers/mdxProvider.tsx (hunk 4 above)
Comment (sohan2410) [reply providers/mdxProvider.tsx:19]: ```className="highlight-box"``` is already written in md files

File: next.dynamic.mjs (hunk 2 above)
Comment (sohan2410) [reply next.dynamic.mjs]: well ```rehypeSlug```  add ids to the headings, ```rehypeAutolinkHeadings``` doesn't work as expected

File: providers/mdxProvider.tsx (hunk 4 above)
Comment (ovflowd) [reply providers/mdxProvider.tsx:19]: Which md files? Still please do not remove this. This is not part of the scope of this PR 🙇

File: next.dynamic.mjs (hunk 2 above)
Comment (ovflowd) [reply next.dynamic.mjs]: Wdym doesn't work as expected? It's a well known plugin. Maybe you're using it incorrectly? Please read the docs ;)

File: providers/mdxProvider.tsx (hunk 4 above)
Comment (ovflowd) [reply providers/mdxProvider.tsx:19]: Friendly reminder to add the `blockquote` entry back.

Even if there are Markdown files already with this classname, the "blockquote" conversion is used for Markdown tags not HTML ones.

**Review by ovflowd — APPROVED** [review]
LGTM! Excellent stuff!

This not only fixes our current headings with wrong ID issues but also automatically adds IDs for the ones that don't have one.

**Pull request merged**